### PR TITLE
feat: add models and contract caller

### PR DESCRIPTION
**Pull request by irrun**

### CODE REVIEW

File: model/model.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	Operator  common.Address `gorm:"column:operator;uniqueIndex:idx_operator"`
+	Consensus common.Address `gorm:"column:consensus"`
+	Credit    common.Address `gorm:"column:credit"`
+}
+
+// Validator is a model for the validator table.
+type Validator struct {
+	ID uint64 `gorm:"column:id;primaryKey"`
+
+	Operator common.Address `gorm:"column:operator;uniqueIndex:idx_operator_date"`
```
Comment (unclezoro) [inline model/model.go:27]: This should not be uniqueIndex, because we will have duplicated record for one validator

File: model/model.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+	// RewardAfterCommission query StakeCredit rewardRecord at the end of the day
+	RewardAfterCommission *Big `gorm:"column:reward_after_commission;type:VARBINARY(32)"`
+	// TotalPooledBNB query StakeCredit totalPooledBNBRecord at the end of the day
+	TotalPooledBNB *Big `gorm:"column:total_pooled_bnb;type:VARBINARY(32)"`
+	// TotalCreditToken query StakeCredit _totalSupply at the previous block of the breathing block
+	TotalCreditToken *Big `gorm:"column:total_credit_token"`
```
Comment (unclezoro) [inline model/model.go]: why this is not VARBINARY(32)?

File: model/model.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+	Date uint64 `gorm:"column:date;uniqueIndex:idx_operator_date"`
```
Comment (unclezoro) [inline model/model.go:35]: should this be unique?

File: model/model.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+	TxHash      common.Hash    `gorm:"column:tx_hash;uniqueIndex:idx_tx_hash"`
+}
+
+// BreathBlockRewardEvent
+// getLogs{toAddress:[credits], topics:RewardReceived}
+type BreathBlockRewardEvent struct {
+	ID uint64 `gorm:"column:id;primaryKey"`
+
+	Credit                common.Address `gorm:"column:operator;uniqueIndex:idx_credit_date"`
+	RewardAfterCommission *Big           `gorm:"column:reward_after_commission;type:VARBINARY(32)"`
```
Comment (unclezoro) [inline model/model.go:78]: I think it is not `RewardAfterCommission `, it is `reward`, actually it is before the commission.

File: model/model.go [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+// getLogs{toAddress:stakeHub, topic:Delegated/Undelegated/Redelegated}.
+type DelegateTx struct {
+	ID uint64 `gorm:"column:id;primaryKey"`
+
+	Delegator   common.Address `gorm:"column:delegator"`
+	SrcOperator common.Address `gorm:"column:src_operator"`
+	DstOperator common.Address `gorm:"column:dst_operator"`
+	Action      DelegateAction `gorm:"column:action"`
+	Amount      *Big           `gorm:"column:amount;type:VARBINARY(32)"`
+	TxHash      common.Hash    `gorm:"column:tx_hash;uniqueIndex:idx_tx_hash"`
```
Comment (unclezoro) [inline model/model.go:68]: add timestamp here

File: model/model.go [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+	Operator common.Address `gorm:"column:operator;index:idx_operator"`
+	Amount   *Big           `gorm:"column:amount;type:VARBINARY(32)"`
+	TxHash   common.Hash    `gorm:"column:tx_hash;uniqueIndex:idx_tx_hash"`
+
+	Date int64 `gorm:"column:date;index:idx_date"`
+}
+
+// Header is a model for the header table.
+type Header struct {
+	ID uint64 `gorm:"column:id;primaryKey" rlp:"-"`
```
Comment (unclezoro) [inline model/model.go]: do we really need this table.. I thought just a table to record to where our service have synced to.

File: model/model.go (hunk 2 above)
Comment (irrun) [reply model/model.go]: done

File: model/model.go (hunk 5 above)
Comment (irrun) [reply model/model.go:68]: done

File: model/model.go (hunk 6 above)
Comment (irrun) [reply model/model.go]: done

**Pull request merged**